Petar_Nedyalkov/week2/task3.cpp

```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+#include <fstream>
+
+struct Pair {
+	unsigned int x;
+	unsigned int y;
+};
+
+Pair createPair(unsigned int first, unsigned int second)
+{
+	Pair newPair;
+	newPair.x = first;
+	newPair.y = second;
+
+	return newPair;
+}
+
+void initPair(Pair& toInit, unsigned int x,unsigned int y)
+{
+	toInit.x = x;
+	toInit.y = y;
+}
+
+Pair readPairFromFile(std::ifstream& inFile)
+{
+	Pair newPair;
+	if (!inFile.is_open())
```

Comment:
If you're not the one opening the file - you shouldn't be checking whether it's open or not. This one is best handled right after opening.